cmd/podman/kill.go

```diff
@@ -37,12 +42,18 @@ var (
 // killCmd kills one or more containers with a signal
 func killCmd(c *cli.Context) error {
 	args := c.Args()
-	if len(args) == 0 && !c.Bool("latest") {
+	if (!c.Bool("all") && !c.Bool("latest")) && len(args) == 0 {
 		return errors.Errorf("specify one or more containers to kill")
 	}
-	if len(args) > 0 && c.Bool("latest") {
+	if (c.Bool("all") || c.Bool("latest")) && len(args) > 0 {
 		return errors.Errorf("you cannot specific any containers to kill with --latest")
```

Comment:
Maybe Saturday brain, but isn't this going to trigger if you have just `podman kill --all`?  I think you can drop the all check here.

Reply:
Line 49, not yours, but "specific" to "specify" please.